04-evm-mapping.md

```diff
@@ -0,0 +1,307 @@
+| 1     | secp256k1 pubkey (account actor)  | Account    | 20 bytes      | 21 bytes    | blake2b-160 hash of secp256k1 pubkey                  | Externally, to refer to an account actor with its pubkey                             | Y       |
+| 2     | aliased universal actor address   | All        | 20 bytes      | 21 bytes    | protocol-derived from relevant cryptographic material | Externally and internally to refer to any actor                                      | Y       |
+| 3     | bls pubkey (account actor)        | Account    | 48 bytes      | 49 bytes    | inlined bls public key                                | Externally, to refer to an account actor with its pubkey                             | Y       |
+| 4     | canonical universal actor address | All        | 32 bytes      | 33 bytes    | protocol-derived from relevant cryptographic material | Externally and internally to refer to any actor                                      | Y       |
+
+## Gas accounting and execution halt semantics
+
+The execution halt is determined by Filecoin gas and not by EVM gas. Therefore, EVM runtime is made to run with unlimited gas. The FVM is responsible for metering execution and halting it when gas limits are exceeded. Refer to the [Gas accounting](01-architecture.md#gas-accounting) section of the Architecture doc for more details.
+
+The exit code of the execution matches the Filecoin value for "out of gas" situations.
+
+## Ethereum logs/events
+
+TBD.
+
+## Blockchain timing
+
+Ethereum target block times are ~10 seconds, whereas Filecoin's is ~30 seconds. A priori, this difference has no impact on the protocol or this spec, but it may impact the behaviour of smart contracts ported over from Ethereum that expect 10-second block timing.
+
+## Precompiles
+
+Precompiles from Ethereum will be honoured.
+
+However, because the full EVM runtime is compiled to WASM, so are the precompiles provided in the source code of the EVM runtime. Therefore, technically speaking, they are not _native precompiles_.
+
+In the future, we may optimise by having them call a Filecoin syscall to escape to native land. This would imply a traversal of Boundary A. See the [Domains and boundaries](01-architecture.md#domains-and-boundaries) section of the architecture doc.
+
+## Cryptographic primitives
+
+TBD.
+
+## Chain-specific opcodes
+
+The following are some notable chain specific opcodes whose behaviour is translated to the Filecoin environment as specified:
+
+* `BLOCKHASH`: returns the hash of the first block in the specified lookback tipset. 
+* `GASLIMIT`: returns the gas limit as per Filecoin gas system.
+* `CHAINID`: returns a fixed value `0`.
+* `GAS`: returns the gas remaining as per Filecoin gas system. One divergence from Ethereum is the return value does not include the full cost of this operation (because the cost of stack copy and program advance is not known when the value is captured).
```

Comment:
Using the FIlecoin gas here diverges from what near does and may cause more problems than it solves because the gas models won't match up.